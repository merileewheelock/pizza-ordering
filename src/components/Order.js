import React, {Component} from 'react';
import $ from 'jquery';

class Order extends Component{
	constructor(props){
		super(props);
		this.state = {
			options: [],
			totalPrice: 0
		}
		this.handleTopping = this.handleTopping.bind(this);
		this.handleSize = this.handleSize.bind(this);
		this.handleButton = this.handleButton.bind(this);
	}

	componentDidMount(){
        $.getJSON('http://localhost:3000/order', (options)=>{
            this.setState({
                options: options
            })
        });
    }

    handleTopping(value, index){
    	console.log($('.topping'))
    	var clickedTopping = $('.topping')

    	if ($(clickedTopping[index]).hasClass('selected')){
    		// topping is already added, don't add again
    		return;
    	}

    	$(clickedTopping[index]).addClass('selected');

    	this.setState({
    		totalPrice: this.state.totalPrice + value
    	})
    }

    handleSize(value, index){

    	var clickedSize = $('.pizza-size');
    	var pizzaArrayIndex = index - 12;

    	if ($(clickedSize[pizzaArrayIndex]).hasClass('selected')){
    		// pizza size already selected, no more choosing a size
    		return;
    	}

    	if (pizzaArrayIndex === 0){
    		$(clickedSize[1]).addClass('off');
    		$(clickedSize[2]).addClass('off');
    	}else if (pizzaArrayIndex === 1){
    		$(clickedSize[0]).addClass('off');
    		$(clickedSize[2]).addClass('off');
    	}else if (pizzaArrayIndex === 2){
    		$(clickedSize[0]).addClass('off');
    		$(clickedSize[1]).addClass('off');
    	}
    	if ($(clickedSize[pizzaArrayIndex]).hasClass('off')){
    		// pizza size already selected, no more choosing a size
    		return;
    	}

    	$(clickedSize[pizzaArrayIndex]).addClass('selected');

    	this.setState({
    		totalPrice: this.state.totalPrice + value
    	})
    }

    handleButton(){
    	this.props.history.push('/submitted');
    }

	render(){

		var toppingsArray = [];
		this.state.options.map((option, index)=>{
			if (option.topping === 1){
				toppingsArray.push(
					<tr key={index} className="topping" onClick={()=>this.handleTopping(option.price, index)}>
						<td>{option.name}</td>
						<td>${option.price.toFixed(2)}</td>
					</tr>
				)
			}
		});

		var sizeArray = [];
		this.state.options.map((option, index)=>{
			if (option.size === 1){
				sizeArray.push(
					<tr key={index} className="pizza-size" onClick={()=>this.handleSize(option.price, index)}>
						<td>{option.name}</td>
						<td>${option.price.toFixed(2)}</td>
					</tr>
				)
			}
		});

		return(
			<div className="order-page container col-sm-6 col-sm-offset-3">
				<div className="row">
					<h1 className="order-header text-center">my pizza</h1>
				</div>

				<div className="row">

					<div className="pizza-size-section col-sm-6">
						<table className="table">
							<thead>
								<tr>
									<th className="text-center">Size</th>
									<th className="text-center">Price</th>
								</tr>
							</thead>
							<tbody className="text-center">
								{sizeArray}
							</tbody>
						</table>

						<div className="total-price row text-center">
							order total: ${this.state.totalPrice.toFixed(2)}
						</div>
						<div className="row text-center">
							<img src="/images/dishes.png" alt="plate" />
						</div>
					</div>

					<div className="col-sm-6">
						<table className="table">
							<thead>
								<tr>
									<th className="text-center">Topping</th>
									<th className="text-center">Price</th>
								</tr>
							</thead>
							<tbody className="text-center">
								{toppingsArray}
							</tbody>
						</table>
					</div>

				</div>

				<div className="row">
					<div className="text-center">
						<button className="order-btn btn btn-default" onClick={this.handleButton}>Order</button>
					</div>
				</div>

			</div>
		)
	}
}

export default Order;