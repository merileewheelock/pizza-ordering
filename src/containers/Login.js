import React, { Component } from 'react';
import  {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import LoginAction from '../actions/LoginAction';

class Login extends Component{
	constructor(props){
		super(props);
		this.state = {
			loginMessage: "",
			formError: false
		}
		this.handleLogin = this.handleLogin.bind(this);
	}

	handleLogin(event){
		event.preventDefault();
		var username = event.target[0].value
		var password = event.target[1].value
		var error = false;

		if(error){
			this.setState({
				formError: true,
			}) 
		}else{
			this.props.loginAction({
				username: username,
				password: password
			});
		}
	}

	componentWillReceiveProps(nextProps) {
		if(nextProps.loginResponse.msg === 'loginSuccess'){
			this.props.history.push('/edit');
		}else if(nextProps.loginResponse.msg === 'badUsername'){
			this.setState({
				loginMessage: "Sorry, that is not a valid username."
			})
		}else if(nextProps.loginResponse.msg === 'wrongPassword'){
			this.setState({
				loginMessage: "Password is incorrect."
			})
		}
	}

	render(){
		return(
			<div className="login container">
				<div className="col-sm-4 col-sm-offset-4">
				<h1 className="login-header text-center">Admin Login</h1>
				<h3 className="login-text text-center">{this.state.loginMessage}</h3>
					<form onSubmit={this.handleLogin}>
						<div className="form-group">
							<label htmlFor="usr" className="login-text">Name:</label>
							<input type="text" className="form-control" id="usr" />
						</div>
						<div className="form-group">
							<label htmlFor="pwd" className="login-text">Password:</label>
							<input type="password" className="form-control" id="pwd" />
						</div>
						<div className="text-center">
							<button className="btn btn-default login-btn" type="submit">Login</button>
						</div>
					</form>
				</div>
			</div>
		)
	}
}

function mapStateToProps(state){
	return{
		loginResponse: state.loginReducer
	}
}

function mapDispatchToProps(dispatch){
	return bindActionCreators({
		loginAction: LoginAction
	}, dispatch)
}

export default connect(mapStateToProps,mapDispatchToProps)(Login);