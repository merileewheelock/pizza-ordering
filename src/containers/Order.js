import React, {Component} from 'react';
import $ from 'jquery';

class Order extends Component{
	constructor(props){
		super(props);
		this.state = {
			options: [],
			totalPrice: 0
		}
		this.handleTopping = this.handleTopping.bind(this);
		this.handleSize = this.handleSize.bind(this);
	}

	componentDidMount(){
        $.getJSON('http://localhost:3000/order', (options)=>{
            this.setState({
                options: options
            })
        });
    }

    handleTopping(value){
    	$(this).addClass("selected");
    	this.setState({
    		totalPrice: this.state.totalPrice + value
    	})
    }

    handleSize(value){
    	this.setState({
    		totalPrice: this.state.totalPrice + value
    	})
    }

	render(){

		var toppingsArray = [];
		this.state.options.map((option, index)=>{
			if (option.topping === 1){
				toppingsArray.push(
					<tr key={index} className="topping" onClick={()=>this.handleTopping(option.price)}>
						<td>{option.name}</td>
						<td>${option.price.toFixed(2)}</td>
					</tr>
				)
			}
		});

		var sizeArray = [];
		this.state.options.map((option, index)=>{
			if (option.size === 1){
				sizeArray.push(
					<tr key={index} className="size" onClick={()=>this.handleSize(option.price)}>
						<td>{option.name}</td>
						<td>${option.price.toFixed(2)}</td>
					</tr>
				)
			}
		});

		return(
			<div className="order-page container col-sm-6 col-sm-offset-3">
				<div className="row">
					<h1 className="order-header text-center">my pizza</h1>
				</div>

				<div className="row">

					<div className="pizza-size col-sm-6">
						<table className="table">
							<thead>
								<tr>
									<th className="text-center">Size</th>
									<th className="text-center">Price</th>
								</tr>
							</thead>
							<tbody className="text-center">
								{sizeArray}
							</tbody>
						</table>

						<div className="total-price row text-center">
							Price: ${this.state.totalPrice.toFixed(2)}
						</div>
					</div>

					<div className="col-sm-6">
						<table className="table">
							<thead>
								<tr>
									<th className="text-center">Topping</th>
									<th className="text-center">Price</th>
								</tr>
							</thead>
							<tbody className="text-center">
								{toppingsArray}
							</tbody>
						</table>
					</div>

				</div>

			</div>
		)
	}
}

export default Order;