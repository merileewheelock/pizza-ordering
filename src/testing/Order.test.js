import React from 'react';
import { shallow } from 'enzyme';
import { handleTopping } from '../containers/Order';

// describe('handleTopping', ()=>{
// 	it('should accept a price value and index', function(){

// 	})
// })
