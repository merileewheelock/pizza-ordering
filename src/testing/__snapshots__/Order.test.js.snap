// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="order-page container col-sm-6 col-sm-offset-3"
>
  <div
    className="row"
  >
    <h1
      className="order-header text-center"
    >
      my pizza
    </h1>
  </div>
  <div
    className="row"
  >
    <div
      className="pizza-size-section col-sm-6"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th
              className="text-center"
            >
              Size
            </th>
            <th
              className="text-center"
            >
              Price
            </th>
          </tr>
        </thead>
        <tbody
          className="text-center"
        />
      </table>
      <div
        className="total-price row text-center"
      >
        order total: $
        0.00
      </div>
      <div
        className="row text-center"
      >
        <img
          alt="plate"
          src="/images/dishes.png"
        />
      </div>
    </div>
    <div
      className="col-sm-6"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th
              className="text-center"
            >
              Topping
            </th>
            <th
              className="text-center"
            >
              Price
            </th>
          </tr>
        </thead>
        <tbody
          className="text-center"
        />
      </table>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="text-center"
    >
      <button
        className="order-btn btn btn-default"
        onClick={[Function]}
      >
        Order
      </button>
    </div>
  </div>
</div>
`;
